File: best-time-to-buy-and-sell-stock.cpp
class Solution {
    public int maxProfit(int[] prices) {
        int currmin=prices[0];
        int ans=0;
        for(int i=1;i<prices.length;i++){
            ans=Math.max(ans,prices[i]-currmin);
            currmin=Math.min(currmin,prices[i]);
        }
        return ans;
    }
}

File: best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int maxiprofit = 0;
        int currmin = prices[0];
        for(int i=1;i<n;i++){
            maxiprofit = Math.max(maxiprofit,prices[i]-currmin);
            currmin = Math.min(currmin,prices[i]);
        }
        return maxiprofit;
    }
}